rfcs/field-selection.md

```diff
@@ -0,0 +1,1453 @@
+     - A `Selection` inherently acts as an `InputField` with the same name as the field it selects:
+       - If multiple field names are possible, the `Selection` must be nested within an `InputField`.
+     - Selecting multiple fields within a single `Selection` is prohibited to prevent ambiguity.
+     - Aliases are not supported within a `Selection`.
+     - Arguments are not supported within a `Selection`.
+     - Directives are not supported within a `Selection`.
+   - An `InputField` corresponds directly to a field within the input object:
+     - It is a key-value pair where the key is the field name and the value is either an `ObjectField` or a `Selection` -> `fieldName: value`.
+       - If the value of an `ObjectField` begins with a `{`, it is classified as an `ObjectValue`.
+       - Otherwise, it is treated as a `Selection`.
+   - An `ObjectValue` is an aggregate of `InputField`s.
+
+So simple field selection would look like this:
+**Single Field Reference**
+```graphql
+extend type Query {
+    userById(id: ID! @is(field: "id")): User @lookup
+}
+```
+
+and multiple field selection would look like this:
+**Multiple Field Reference**
+```graphql
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName lastName")): User @lookup
+}
+```
+
+if you need to rename a field you just use a `InputField` and set the value to an `ObjectField`
+
+**Multiple Field Reference - Fieldname is different**
+```graphql
+extend type Query {
+    #                   Selection:                         +--------+           +---------+
+    #                   InputField:            +---------+          +---------+
+    findUserByName(user: UserInput! @is(field: "fristName: givenName lastName: familyName")): User @lookup
+}
+```
+
+You can easily express abstract types as well
+
+**Abstract Field Reference**
+```graphql
+extend type Query {
+   mediaById(id: ID! @is(field: "... Movie { id } ... Book { id }")): MovieOrBook @lookup
+}
+```
+
+but you can also shape the input object
+
+**Multiple Field Reference - Output field is deeper in the tree**
+```graphql
+type Query {
+    #
+    #                       Selection:                    +--------------------+ +--------+
+    #                       InputField:        +---------+
+    findUserByName(user: UserInput! @is(field: "firstName: profile { firstName } lastName")): User @lookup
+}
+```
+
+you can use abstract types as well
+
+**Multiple Field Reference - Output field is deeper in the tree with abstract type and fieldname is different**
+```graphql
+    #
+    #                       Selection:                    +------------------------------------------------------------------------+ +--------+
+    #                       InputField:        +---------+
+    findUserByName(user: UserInput! @is(field: "firstName: profile { ... on EntraProfile { name } ... on AdfsProfile { firstName } } lastName")): User @lookup
```

Comment:
There's long been open a proposal for [flat chain syntax](https://github.com/graphql/graphql-spec/issues/174) and I think Lee even mentioned he'd be interested in seeing this. This feels like a compelling use-case to get such a syntax adopted into the spec. Handling polymorphism is something that that particular RFC doesn't address AFAIK. Perhaps a `.` would be enough:

```suggestion
    findUserByName(user: UserInput! @is(field: "firstName: profile.{ ... on EntraProfile { name } ... on AdfsProfile { firstName } } lastName")): User @lookup
```

This would also improve the previous example:

```diff
-    findUserByName(user: UserInput! @is(field: "firstName: profile { firstName } lastName")): User @lookup
+    findUserByName(user: UserInput! @is(field: "firstName: profile.firstName lastName")): User @lookup
```